### ISSUE TITLE
conda run doesn't forward the exit code

### ISSUE THREAD

**Issue by IRDonch** [conda/conda#8385]
## Current Behavior

The return code of `conda run` is always 0, no matter what the executed program returns.

### Steps to Reproduce

```
>conda run -n base cmd /c exit 42
ERROR conda.cli.main_run:execute(26): Subprocess for 'conda run' command failed.  Stderr was:
The system cannot find the path specified.


>echo %errorlevel%
0
```
(Note also the strange "The system cannot find the path specified" message from `conda run` - that seems like it shouldn't be there.)

## Expected Behavior

`echo %errorlevel%` should have output "42".

## Environment Information
<details open><summary><code>`conda info`</code></summary><p>

```
     active environment : base
    active env location : C:\Users\[redacted]\AppData\Local\Continuum\miniconda3
            shell level : 1
       user config file : C:\Users\[redacted]\.condarc
 populated config files :
          conda version : 4.6.7
    conda-build version : not installed
         python version : 3.7.1.final.0
       base environment : C:\Users\[redacted]\AppData\Local\Continuum\miniconda3  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/win-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/win-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/win-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/msys2/win-64
                          https://repo.anaconda.com/pkgs/msys2/noarch
          package cache : C:\Users\[redacted]\AppData\Local\Continuum\miniconda3\pkgs
                          C:\Users\[redacted]\.conda\pkgs
                          C:\Users\[redacted]\AppData\Local\conda\conda\pkgs
       envs directories : C:\Users\[redacted]\AppData\Local\Continuum\miniconda3\envs
                          C:\Users\[redacted]\.conda\envs
                          C:\Users\[redacted]\AppData\Local\conda\conda\envs
               platform : win-64
             user-agent : conda/4.6.7 requests/2.21.0 CPython/3.7.1 Windows/10 Windows/10.0.16299
          administrator : False
             netrc file : None
           offline mode : False
```
</p></details>

<details open><summary><code>`conda config --show-sources`</code></summary><p>

```

```
</p></details>

<details><summary><code>`conda list --show-channel-urls`</code></summary><p>

```
# packages in environment at C:\Users\[redacted]\AppData\Local\Continuum\miniconda3:
#
# Name                    Version                   Build  Channel
asn1crypto                0.24.0                   py37_0    defaults
ca-certificates           2019.1.23                     0    defaults
certifi                   2018.11.29               py37_0    defaults
cffi                      1.11.5           py37h74b6da3_1    defaults
chardet                   3.0.4                    py37_1    defaults
conda                     4.6.7                    py37_0    defaults
conda-env                 2.6.0                         1    defaults
console_shortcut          0.1.1                         3    defaults
cryptography              2.4.2            py37h7a1dbc1_0    defaults
idna                      2.8                      py37_0    defaults
menuinst                  1.4.14           py37hfa6e2cd_0    defaults
openssl                   1.1.1b               he774522_0    defaults
pip                       18.1                     py37_0    defaults
pycosat                   0.6.3            py37hfa6e2cd_0    defaults
pycparser                 2.19                     py37_0    defaults
pyopenssl                 18.0.0                   py37_0    defaults
pysocks                   1.6.8                    py37_0    defaults
python                    3.7.1                h8c8aaf0_6    defaults
pywin32                   223              py37hfa6e2cd_1    defaults
requests                  2.21.0                   py37_0    defaults
ruamel_yaml               0.15.46          py37hfa6e2cd_0    defaults
setuptools                40.6.3                   py37_0    defaults
six                       1.12.0                   py37_0    defaults
sqlite                    3.26.0               he774522_0    defaults
urllib3                   1.24.1                   py37_0    defaults
vc                        14.1                 h0510ff6_4    defaults
vs2015_runtime            14.15.26706          h3a45250_0    defaults
wheel                     0.32.3                   py37_0    defaults
win_inet_pton             1.0.1                    py37_1    defaults
wincertstore              0.2                      py37_0    defaults
yaml                      0.1.7                hc54c509_2    defaults
```
</p></details>

**Comment by kevinblack100** [comment]
Regarding the message "The system cannot find the path specified", it seems like `conda run` is trying to interpret the error code or standard error somehow. To see this,

First, create `hello_only.py` including
```python
print('Hello World')
```

Then, invoke `conda run`
```
(base) >conda run -n base python hello_only.py
Hello World

(base) >echo %errorlevel%
0
```

No problem there. But now, create `hello_and_goodbye.py` including
```python
import sys
print('Hello World')
sys.exit('Goodbye')
```

Then, invoke `conda run`
```
(base) >conda run -n base python hello_and_goodbye.py
Hello World
ERROR conda.cli.main_run:execute(26): Subprocess for 'conda run' command failed.  Stderr was:
The system cannot find the path specified.
Goodbye

(base) >echo %errorlevel%
0
```
It should have had the error code `1`, but more importantly note that the message "The system cannot find the path specified" seemingly appears in between the "Hello World" and "Goodbye" messages.

`conda info`
```
     active environment : base
    active env location : C:\Users\[redacted]\miniconda
            shell level : 1
       user config file : C:\Users\[redacted]\.condarc
 populated config files : C:\Users\[redacted]\.condarc
          conda version : 4.6.8
    conda-build version : not installed
         python version : 3.7.1.final.0
       base environment : C:\Users\[redacted]\miniconda  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/win-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/win-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/win-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/msys2/win-64
                          https://repo.anaconda.com/pkgs/msys2/noarch
          package cache : C:\Users\[redacted]\miniconda\pkgs
                          C:\Users\[redacted]\.conda\pkgs
                          C:\Users\[redacted]\AppData\Local\conda\conda\pkgs
       envs directories : C:\Users\[redacted]\a_apps\miniconda\envs
                          C:\Users\[redacted]\.conda\envs
                          C:\Users\[redacted]\AppData\Local\conda\conda\envs
               platform : win-64
             user-agent : conda/4.6.8 requests/2.21.0 CPython/3.7.1 Windows/10 Windows/10.0.16299
          administrator : False
             netrc file : None
           offline mode : False
```

**Comment by IRDonch** [comment]
Tried this again with conda 4.7.12. The mysterious message is no longer displayed; however, the exit code is still not forwarded.

**Comment by jaredjennings** [comment]
It appears as though commit 06030b3ea794eb243ae7312fa81da65681ab6f2f is a response to this problem. If I'm reading the GitHub page right, that code has been released in conda 4.8.0rc0 on Nov 4. That's a prerelease, and I'm not sure yet how to move to it. But, failing everything else, it will be released at some time in the future.

**Comment by ehmoussi** [comment]
The problem still exists even with the 06030b3 modifications. I use conda 4.8.2 and the exit code is always 0 with the run command.

**Comment by aachkar-miovision** [comment]
I can confirm this issue with a simple reproduction:


Conda 4.7.12 - works as expected
```
aachkar$ docker run continuumio/miniconda3:4.7.12 /bin/bash -c "conda run exit 1; echo rc: $?"
ERROR conda.cli.main_run:execute(39): Subprocess for 'conda run ['exit', '1']' command failed.  Stderr was:

rc: 1
```

Conda 4.8.2 - swallows up return code
```
aachkar$ docker run continuumio/miniconda3:4.8.2 /bin/bash -c "conda run exit 1; echo rc: $?"
ERROR conda.cli.main_run:execute(32): Subprocess for 'conda run ['exit', '1']' command failed.  (See above for error)
rc: 0
```

**Comment by aachkar-miovision** [comment]
I believe the issue is that https://github.com/conda/conda/blob/b200354f6358696fc25e382ca208daaff3f59d24/conda/cli/main_run.py#L42 returns a response whereas main only returns an exit code if the return of the function execution returns an integer: https://github.com/conda/conda/blob/b200354f6358696fc25e382ca208daaff3f59d24/conda/cli/main.py#L84-L86

**Comment by analog-cbarber** [comment]
FYI, the corresponding bug has been fixed in mamba 0.4.2, so you can use `mamba run` to work around this bug.

**Comment by RoelantStegmann** [comment]
Also having this issue, which made my build pass when my tests were failing!

**Comment by chenghlee** [comment]
This issue was fixed in conda 4.8.4 (PR #9665).

**Comment by IRDonch** [comment]
I verified with 4.9.1 and it does appear to be fixed. Closing.
